### PR TITLE
[Modules] Updated Authorization/RoleDefinitions to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Authorization/RoleDefinitions to new dependency approach

| Pipeline |
| - |
| [Authorization: RoleDefinitions](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.roledefinitions.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

**Comment by ahmadabdalla** [comment]
Pending change to the implementation approach:
MicrosoftTeams-image

**Comment by MrMCake** [comment]
> Pending change to the implementation approach: MicrosoftTeams-image

As per our discussion I'd suggest to push this back as neither the global.module.tests, nor especially the Set-ModuleReadMe function will be able to properly handle this alternative approach. A larger refactoring may be required.

File: modules/Microsoft.Authorization/roleDefinitions/.test/mg.common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,31 @@
+targetScope = 'managementGroup'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ardmg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleDefinitions/.test/mg.common/deploy.test.bicep:7]:
```suggestion
param serviceShort string = 'ardmgcom'
```

File: modules/Microsoft.Authorization/roleDefinitions/.test/rg.common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.authorization.roledefinitions-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ardrg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleDefinitions/.test/rg.common/deploy.test.bicep:14]:
```suggestion
param serviceShort string = 'ardrgcom'
```

File: modules/Microsoft.Authorization/roleDefinitions/.test/sub.common/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ardsub'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleDefinitions/.test/sub.common/deploy.test.bicep:7]:
```suggestion
param serviceShort string = 'ardsubcom'
```

File: modules/Microsoft.Authorization/roleDefinitions/.test/mg.common/deploy.test.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/roleDefinitions/.test/mg.common/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/roleDefinitions/.test/sub.common/deploy.test.bicep (hunk 3 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/roleDefinitions/.test/sub.common/deploy.test.bicep]: done

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**